### PR TITLE
Implement applyMiddlewares

### PR DESCRIPTION
**Pull request by mkaulig**
Implement a method applyMiddlewares inspired by the redux function with
the same name. This method returns an enhancer, which can enhance store
creators to execute the applied middlewares before normal dispatch
calls.

### CODE REVIEW

File: lib/src/main/java/com/glung/redux/Middlewares.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        }
+
+        @Override
+        public void replaceReducer(Reducer<S> reducer) {
+            nextStore.replaceReducer(reducer);
+        }
+
+        @Override
+        public Object dispatch(Object action) {
+            return createMiddlewareDispatcher(middlewares).dispatch(action);
```
Comment (glung) [inline lib/src/main/java/com/glung/redux/Middlewares.java]: Do you think we could create the Dispatcher it only once by calling `createMiddlewareDispatcher` in `applyMiddlewares` ?

**Review by glung — CHANGES_REQUESTED** [review]
Sorry totally missed your PR ! Thx for contributing 

Could you add tests ?

File: lib/src/main/java/com/glung/redux/Middlewares.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        }
+
+        private Dispatcher createMiddlewareDispatcher(final List<redux.api.enhancer.Middleware<S>> middlewares) {
```
Comment (AngusMorton) [inline lib/src/main/java/com/glung/redux/Middlewares.java]: This method should be static

**Comment by mkaulig** [comment]
Thanks for the feedback! I added some Tests and changed the implementation to create the ```Dispatcher``` only once in ```applyMiddlewares```

**Comment by glung** [comment]
Hey @mkaulig just a message to say I am not ignoring this PR, it will look into it asap

**Review by glung — APPROVED** [review]
Thx! I am just going to apply some minor style related changes on master after landing it

**Pull request merged**